pkg/platform/kube/functionres/lazy.go

```diff
@@ -195,6 +196,10 @@ func (lc *lazyClient) CreateOrUpdate(ctx context.Context, function *nuclioio.Nuc
 		}
 	}
 
+	if err := lc.validateFunction(function); err != nil {
```

Comment:
+1 for recovering controller but I think it should also call the validation so it would have consistent validation and logging with the dashboard / nuctl

Reply:
I moved the validations to platform, while preventing the controller from crashing (by adding error handling on GetIngressesFromTriggers)